File: services/reservation/src/main/java/com/example/iter/reservation/dto/response/UserReviewStatsResponse.java
package com.example.iter.reservation.dto.response;

import com.example.iter.reservation.domain.repository.RentalReviewRepository;

public record UserReviewStatsResponse(
        double averageRating,
        long reviewCount
) {
    public static UserReviewStatsResponse from(RentalReviewRepository.RatingStats stats) {
        if (stats == null || stats.getReviewCount() == null || stats.getReviewCount() == 0) {
            return new UserReviewStatsResponse(0.0, 0L);
        }
        return new UserReviewStatsResponse(stats.getAverageRating(), stats.getReviewCount());
    }
}

File: services/reservation/src/main/kotlin/com/example/iter/reservation/dto/response/UserReviewStatsResponse.kt
package com.example.iter.reservation.dto.response

import com.example.iter.reservation.domain.repository.RentalReviewRepository

data class UserReviewStatsResponse(
    val averageRating: Double,
    val reviewCount: Long,
) {
    companion object {
        @JvmStatic
        fun from(stats: RentalReviewRepository.RatingStats?): UserReviewStatsResponse {
            if (stats == null || stats.reviewCount == null || stats.reviewCount == 0L) {
                return UserReviewStatsResponse(0.0, 0L)
            }
            return UserReviewStatsResponse(stats.averageRating, stats.reviewCount)
        }
    }
}
